gui/elements/qt/_control_bars.py

```diff
@@ -0,0 +1,59 @@
+from PyQt5.QtWidgets import QHBoxLayout, QWidget
+from PyQt5.QtCore import QSize
+
+from ._range_slider import QVRangeSlider
+
+
+class QtControlBars(QWidget):
+    def __init__(self, control_bars):
+        super().__init__()
+
+        self.control_bars = control_bars
+
+        layout = QHBoxLayout()
+
+        # Clim Slider
+        self.climSlider = QVRangeSlider(slider_range=[0, 1, 0.0001],
+                                        values=[0, 1], parent=self)
+        self.climSlider.setEmitWhileMoving(True)
+        self.climSlider.collapsable = False
+        layout.addWidget(self.climSlider)
+
+        self.setLayout(layout)
+        self.setMouseTracking(True)
+        self.setMinimumSize(QSize(40, 40))
+
+        self.climSlider.rangeChanged.connect(self.clim_slider_changed)
+        self.control_bars.events.update_slider.connect(self.update)
+
+    def update(self, event):
+        if event.enabled:
+            self.climSlider.setEnabled(True)
+            self.climSlider.setValues(event.values)
+        else:
+            self.climSlider.setEnabled(False)
+
+    def clim_slider_changed(self, slidermin, slidermax):
+        self.control_bars.values = (slidermin, slidermax)
+        msg = None
+        for layer in self.control_bars.viewer.layers:
+            if hasattr(layer, 'visual') and layer.selected:
+                valmin, valmax = layer._clim_range
+                cmin = valmin+self.control_bars.values[0]*(valmax-valmin)
+                cmax = valmin+self.control_bars.values[1]*(valmax-valmin)
+                layer.clim = [cmin, cmax]
+                msg = '(%.3f, %.3f)' % (cmin, cmax)
```

Comment:
```suggestion
                msg = f'({cmin:0.3}, {cmax:0.3})'
```

Reply:
Hmm, can't say I love how this is treating the numbers for the case of a 0-255 image. The cmin and cmax can be floats, but it is printing them like `(0.0, 2.55e+02)`, which I don't find very nice to read. Any suggestions on how to improve?

Reply:
Gah, you are right. I suggest we write a function for formatting, because I haven't found an expression that's good in all cases:

- integer values
- floating point values in [0, 1]
- tiny floating point values
- floating point values in [0, 255]
- huge floating point values

```python
In [1]: value = 255

In [3]: f'{value:03}'
Out[3]: '255'

In [4]: value = 255.0

In [5]: f'{value:03}'
Out[5]: '255.0'

In [6]: value = 0.255354

In [7]: f'{value:03}'
Out[7]: '0.255354'

In [8]: f'{value:0.3}'
Out[8]: '0.255'

In [9]: value = 255

In [11]: f'{value:0.3}'
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-11-560db1fe8940> in <module>()
----> 1 f'{value:0.3}'

ValueError: Precision not allowed in integer format specifier

In [12]: value = 255.0

In [13]: f'{value:0.3}'
Out[13]: '2.55e+02'

In [14]: value = 2542132

In [16]: f'{value/1:0.3}'
Out[16]: '2.54e+06'

In [17]: f'{value:03}'
Out[17]: '2542132'
```

Reply:
hmm yup, that was the behavior I observed. Maybe for now we leave it and then revisit with a fix later?